### ISSUE TITLE
Python 3.6 in macOS 11, tkinter error: macOS 11 (1107) or later required, have instead 11 (1106)

### ISSUE THREAD

**Issue by carlosperate** [actions/setup-python#580]
**Description:**
Looks like Tkinter from Python 3.6 is not compatible with the current `macos-11` runner.

**Action version:**
v4, which at the time of writting is v4.4.0

**Platform:**
- [ ] Ubuntu
- [x] macOS
- [ ] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

**Tools version:**
- Python 3.6
    - 3.6.15, 3.6.7, 3.6.8, 3.6.10 and others I've tested but didn't write down
- Current runner version: '2.299.1'

**Repro steps:**  

This simple script opens a tkinter window and closes it after 5 seconds:
```python
import tkinter as tk

root = tk.Tk()
label = tk.Label(root, text="Hello World!")
label.pack(padx=20, pady=20)
root.after(5000, root.destroy)    # Close the Window after 5 seconds
root.mainloop()
```

With this workflow:

```yml
name: Test Python Tk

on: [push, pull_request]

jobs:
  tests-poetry:
    strategy:
      matrix:
        os: [ubuntu-20.04, macos-11, windows-latest]
        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
      fail-fast: false
    name: Py ${{ matrix.python-version }} - ${{ matrix.os }}
    runs-on: ${{ matrix.os }}
    steps:
      - uses: actions/checkout@v3
      - name: Set up Python ${{ matrix.python-version }}
        uses: actions/setup-python@v4
        with:
          python-version: ${{ matrix.python-version }}
      - name: Prepare Ubuntu xvfb
        if: runner.os == 'Linux'
        run: |
          sudo apt-get update
          sudo apt-get install -y libxkbcommon-x11-0 xvfb
      - name: Run tk script (closes itself after 5 seconds) - Ubuntu
        if: runner.os == 'Linux'
        run: xvfb-run python tk_min.py
      - name: Run tk script (closes itself after 5 seconds)
        if: runner.os != 'Linux'
        run: python tk_min.py
```

Works in the other Python versions and operating systems except 3.6 on macOS 11:
- https://github.com/carlosperate/pyinstaller-test/actions/runs/3849775929/jobs/6559162909

<img width="1506" alt="image" src="https://user-images.githubusercontent.com/4189262/210865083-823822bd-1e38-4577-90b4-3f97f9b2498b.png">

**Expected behavior:**
The Python script to run without Python aborting.

**Actual behavior:**

Python aborts:
```
Run python tk_min.py
  python tk_min.py
  shell: /bin/bash -e {0}
  env:
    pythonLocation: /Users/runner/hostedtoolcache/Python/3.6.15/x64
    PKG_CONFIG_PATH: /Users/runner/hostedtoolcache/Python/3.6.15/x64/lib/pkgconfig
    Python_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.6.15/x64
    Python2_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.6.15/x64
    Python3_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.6.15/x64
macOS 11 (1107) or later required, have instead 11 (1106) !
/Users/runner/work/_temp/465490bc-d5c3-4408-957f-f9a46c582a34.sh: line 1:  2417 Abort trap: 6           python tk_min.py
Error: Process completed with exit code 134.
```

**Comment by MaksimZhukov** [comment]
Hello @carlosperate ! Thank you for reporting this!
We will investigate the issue and get back to you as soon as we have updates.

**Comment by MaksimZhukov** [comment]
Hello @carlosperate! Thank you for your patience!

In accordance with the [Python documentation](https://www.python.org/download/mac/tcltk/), the Python macOS installers for Python 3.6.x dynamically link to Tcl/Tk 8.5 frameworks. It is also mentioned that Tcl/Tk 8.5 has serious bugs that can cause application crashes and If you wish to use Tkinter, you should install and use a newer version of Python that supplies or links with a newer version of Tcl/Tk. 

It seems to me that the problem is related to the compatibility of macOS 11 and Tkinter 8.5. Unfortunately, I guess this won't be fixed since Python 3.6 has reached [EOL](https://devguide.python.org/versions/#unsupported-versions). As a workaround you can use the `macos-12` runner image or Python version that supplies a newer version of Tcl/Tk (`3.7`, `3.8` and etc).

**Comment by MaksimZhukov** [comment]
I am closing the issue. Please let us know if you have any questions or objections.
Thanks!
